### PR TITLE
Add virtual table os_version for windows

### PR DESCRIPTION
**Pull request by ExtremeGTX**
Add os_version virtual table for Windows

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

File: osquery/tables/system/windows/os_version.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+*/
+
+#include "osquery/core/conversions.h"
+#include "osquery/core/windows/wmi.h"
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+#define WMI_QUERY "SELECT CAPTION,VERSION FROM Win32_OperatingSystem"
```
Comment (PoppySeedPlehzr) [inline osquery/tables/system/windows/os_version.cpp]: Is there any particular reason you have this as a `#define`? With most of the WMI queries we've been using `std::string` defined next to the Request. Do you mind moving this to right before you make your request obj?

File: osquery/tables/system/windows/os_version.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  std::vector<WmiResultItem>& wmiResults = wmiRequest.results();
+
+  if (wmiResults.size() == 0) {
+    return {};
+  }
+
+  Row r;
+  std::string version_string;
+
+  wmiResults[0].GetString("Caption", r["name"]);
```
Comment (PoppySeedPlehzr) [inline osquery/tables/system/windows/os_version.cpp]: What happens if `wmiResults` is empty? Mind throwing an `if (!wmiResults.empty())` check here?

File: osquery/tables/system/windows/os_version.cpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  wmiResults[0].GetString("Version", version_string);
+  auto version = osquery::split(version_string, ".");
+
+  switch (version.size()) {
```
Comment (PoppySeedPlehzr) [inline osquery/tables/system/windows/os_version.cpp]: No `default` case? What happens if version ends up being empty? I think here you could either throw in a check on the size of `version` as above with `wmiResults`, or maybe you could have a `default` case that just breaks?

**Comment by PoppySeedPlehzr** [comment]
ok to test

**Comment by osqueryer** [comment]
:-1: The commit 5c225d0a5e02c47089fd726d72dcda89cf0fab9a (Job results: [431](https://jenkins.osquery.io/job/osqueryPullRequestBuild/431/)) failed one or more tests (**Windows**).

**Comment by PoppySeedPlehzr** [comment]
Looks like that test fail might not be related to your code. Once you've addressed the issues and re-pushed your changes the tests will re-run and hopefully we'll see successes :)

File: osquery/tables/system/windows/os_version.cpp (hunk 2 above)
Comment (PoppySeedPlehzr) [inline osquery/tables/system/windows/os_version.cpp]: Nevermind!  I didn't see the check above :)

File: osquery/tables/system/windows/os_version.cpp (hunk 1 above)
Comment (ExtremeGTX) [inline osquery/tables/system/windows/os_version.cpp]: ok, the macro `#define WMI_QUERY` will be changed to `const std::string kWmiQuery`

File: osquery/tables/system/windows/os_version.cpp (hunk 3 above)
Comment (ExtremeGTX) [inline osquery/tables/system/windows/os_version.cpp]: I can't imagine a scenario where i will get empty version, but default case is always considered as good practice

**Comment by PoppySeedPlehzr** [comment]
Table in action. Nice work @ExtremeGTX!
osquery_win10

**Comment by ExtremeGTX** [comment]
This should close #2564

**Pull request merged**